src/AbstractValidatedRequest.php

```diff
@@ -55,37 +46,54 @@ public function isValid(): bool
         return $this->isValid;
     }
 
+    /**
+     * @throws BadRequestException|InvalidRuleException
+     * @internal invoked by RequestValidationSubscriber
+     */
+    final public function validate(): ?Response
+    {
+        $violationList = $this->validator->validate($this->request, $this->constraintFactory->createConstraint($this->getValidationRules()));
+        if (count($violationList) > 0) {
+            return $this->handleViolations($violationList);
+        }
+
+        $response = $this->validateCustomRules();
+        if ($response instanceof Response) {
+            return $response;
+        }
+
+        $this->isValid = true;
+
+        return null;
+    }
+
     /**
      * Get all the constraints for the current query params
      */
     abstract protected function getValidationRules(): ValidationRules;
```

Comment:
@frankdekker 
Can we make this methode return ?ValidationRules? And handle that in validate().
For cases where we only need to do custom validation, and no request-structure validation.